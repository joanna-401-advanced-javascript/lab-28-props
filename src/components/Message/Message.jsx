import React from 'react';
import Book from '../Book/Book';

export default class extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      text: 'Hello World!',
    };
  }

  render(){
    return (
      <div>
        <p>{this.state.text}</p>
        <p>{this.props.headline}</p>
        <Book />
      </div>
    )
  }
}
