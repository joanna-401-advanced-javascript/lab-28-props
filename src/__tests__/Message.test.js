import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../components/Message/Message';

describe('<Message />', () => {
  it('is rendered at application start', () => {
    const app = shallow(<Message />);
    expect(app.find('div').exists()).toBe(true);
    expect(app.find('p').exists()).toBe(true);
  });

  it('state contains headline', () => {
    const message = mount(<Message />);
    expect(message.state('text')).toEqual('Hello World!');
  });

  it('state does not include anything from other components in its state', () => {
    const message = mount(<Message />);
    expect(message.state('headline')).toEqual(undefined);
    expect(message.state('author')).toEqual(undefined);
  });

  it('it can access props from parent', () => {
    const message = mount(<Message />);
    expect(message.props()).toEqual({});
  });

  it('snapshot is rendered correctly', () => {
    const message = renderer
      .create(<Message />)
      .toJSON();
    expect(message).toMatchSnapshot();
  });
});
