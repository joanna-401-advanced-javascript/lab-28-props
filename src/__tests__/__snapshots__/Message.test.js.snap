// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> snapshot is rendered correctly 1`] = `
<div>
  <p>
    Hello World!
  </p>
  <p />
  <div>
    <p>
      Author: 
      Margaret Atwood
    </p>
    <p>
      Title: 
      Oryx and Crake
    </p>
  </div>
</div>
`;
